test/serializer/abstract/ModelFunction.js

```diff
@@ -0,0 +1,15 @@
+// add at runtime: global.fun = console.log;
+var result = [];
+global.log = function (arg) { result.push(arg); };
+if (global.__assumeDataProperty) {
+  __assumeDataProperty(global, "fun", 
```

Comment:
To a reader of the code, it's not clear that "DEFINED" influences the invariant generation. If we stick to a simple enum, then the name should be more descriptive.